File: week-02/day-03/is_in_list.py

# Check if list contains all of the following elements: 4,8,12,16
# Create a function that accepts list_of_numbers as an input
# it should return "True" if it contains all, otherwise "False"

list_of_numbers = [2, 4, 6, 8, 10, 12, 14, 16]

def is_in_list(alist):
    list_of_elements = [4, 8, 12, 15]
    for i in list_of_elements:
        if not i in alist:
            return False
    return True

print(is_in_list(list_of_numbers))

File: week-07/day-2/is-in-list.js
'use strict';
// Check if array contains all of the following elements: 4,8,12,16
// Create a 'numChecker' function that accepts 'listOfNumbers' as an input
// it should return "true" if it contains all, otherwise "false"

var listOfNumbers = [2, 4, 6, 8, 10, 12, 14, 16];

function numChecker(list){
  let numbers = [4,8,12,16];
  numbers = numbers.map(number => list.indexOf(number) !== -1);
  return numbers.indexOf(false) === -1;
}

console.log(numChecker(listOfNumbers));